mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -959,21 +1023,75 @@ func expandAutoHealingPolicies(configured []interface{}) []*compute.InstanceGrou
 	return autoHealingPolicies
 }
 
-func expandStatefulPolicy(configured []interface{}) *compute.StatefulPolicy {
+func expandStatefulPolicy(d *schema.ResourceData) *compute.StatefulPolicy {
+	preservedState := &compute.StatefulPolicyPreservedState{
+	}
+	stateful_disks := d.Get("stateful_disk").(*schema.Set).List()
 	disks := make(map[string]compute.StatefulPolicyPreservedStateDiskDevice)
-	for _, raw := range configured {
+	for _, raw := range stateful_disks {
 		data := raw.(map[string]interface{})
 		disk := compute.StatefulPolicyPreservedStateDiskDevice{
 			AutoDelete: data["delete_rule"].(string),
 		}
 		disks[data["device_name"].(string)] = disk
 	}
-	if len(disks) > 0 {
-		return &compute.StatefulPolicy{PreservedState: &compute.StatefulPolicyPreservedState{Disks: disks}}
+	preservedState.Disks = disks
+	var change bool
+	change = false
+	<% unless version == "ga" -%>
+	if d.HasChange("stateful_internal_ip") {
+		oldInternalIps, newInternalIps := d.GetChange("stateful_internal_ip")
+		if len(newInternalIps.([]interface{})) == 0 {
+			preservedState.NullFields = append(preservedState.NullFields, "internalIPs")
```

Comment:
I did the changes and did some ForceSendFields to delete the fields. As far as now, the disk could be deleted, but not the stateful Ips. https://paste.googleplex.com/6460650149642240 

According to the log above, the patch request is sent, the disk is deleted, not the Ips. My suggestion is to add the Ips in the verifyStateIgnore. And not mark these fields as Computed

Reply:
Just to have it written here as well - this seems like an API bug because:

- Sending empty objects deletes disks but not ips - we'd expect it to be consistent
- The end state of the API object is the same as what was set by the test - it doesn't seem to be getting reset to a default state by the API (so it's probably not an optional + computed field)

Since it's a bug in the API we can't just ignore the read value - that would mean that users think a change has gone through but actually the value on the API side is different from what's in Terraform.